Public/Src/Cache/MemoizationStore/Library/Service/LocalCacheServer.cs

```diff
@@ -42,15 +42,9 @@ public class LocalCacheServer : LocalContentServerBase<ICache, ICacheSession>
         : base(logger, fileSystem, scenario, cacheFactory, localContentServerConfiguration)
         {
             var storesByName = new Dictionary<string, IContentStore>();
-            foreach (var kvp in localContentServerConfiguration.NamedCacheRoots)

```

Comment:
After this change, any stores in localContentServerConfiguration.NamedCacheRoots will be ignored. What if they are different than StoresByName? What if someone tries to intialize differently?

Reply:
And what if someone else has already added stores to that dictionary, which they can do because it is visible to other inheritors and indeed to everyone in assembly.

---
In reply to: [321938661](https://github.com/microsoft/BuildXL/pull/845#discussion_r321938661) [](ancestors = 321938661)

Reply:
I have made StoresByName a protected only variable; I don't know why it was internal in the first place.

---
In reply to: [321938831](https://github.com/microsoft/BuildXL/pull/845#discussion_r321938831) [](ancestors = 321938831,321938661)